### ISSUE TITLE
binary packages do not work on GLIBC_2.14

### ISSUE THREAD

**Issue by rainwoodman** [bccp/conda-channel-bccp#5]
This is a known issue for builds from Travis. The OS on Travis is slightly newer than most super computing centres, making this a concern.

For example, when I installed bigfile on BlueWaters the binary didn't work with error
```
(root) yfeng1@h2ologin2:~/BWSim-PFFT/etc> !hdf2
hdf2bigfile 
Traceback (most recent call last):
  File "/u/sciteam/yfeng1/anaconda3/bin/hdf2bigfile", line 4, in <module>
    import bigfile
  File "/u/sciteam/yfeng1/anaconda3/lib/python3.5/site-packages/bigfile/__init__.py", line 3, in <module>
    from .pyxbigfile import BigFileError, BigFileClosedError, BigBlockClosedError
ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /u/sciteam/yfeng1/anaconda3/lib/python3.5/site-packages/bigfile/pyxbigfile.cpython-35m-x86_64-linux-gnu.so)
```

If this happens, we advice the users revert to use pip install. 

The issue is workedaround at NERSC because at NERSC we build nightlies on the NERSC systems.

**Comment by rainwoodman** [comment]
This is no longer an issue on Edison and Cori. Cray has updated to a never version of glibc .

I realized many systems actually do use GLIBC_2.12 still. A probable work around is 

http://web.archive.org/web/20160107032111/http://www.trevorpounds.com/blog/?p=103

**Comment by rainwoodman** [comment]
@nickhand  I am working something out with classylss at first. I tagged a C compiler only release for testing out my solution.

**Comment by rainwoodman** [comment]
I've developed a hack, and we still have to work on these packages:

[x] bigfile
[x] classylss
[ ] kdcount
[ ] pmesh
[ ] mpsort
[ ] corrfunc
[ ] fitsio
[ ] halotools
[ ] pfft-python

The idea is to forcely include a header file that pins the symbols to versions of glibc routines <=2.12. To identify bad routines, use the command

```
objdump -T extension_so_file |grep GLIBC
```
Then look for entries greater than 2.12.

To find the corresponding version in libc / libm, use
```
objdump -T /lib64/libc.so.? | grep symbol_name
```
Replace libc with libm or change the exact file name / location as needed.

**Comment by rainwoodman** [comment]
Once these are done I will file an issue 'Waiting for the death of GLIBC 2.12 systems'. When those systems are dead for good we can eliminate these abi version pinning hacks.

**Comment by rainwoodman** [comment]
This is solved  b34d92d .
